src/hooks/users/useRecentLocationUsers.ts

```diff
@@ -47,17 +49,41 @@ export const useRecentLocationUsers = (
       if (!worldUsers || !worldUserLocationsById || !locationName)
         return { isSuccess, recentLocationUsers: [] };
 
-      const recentLocationUsers = worldUsers.filter((user) => {
-        const userLocation: WithId<UserLocation> | undefined =
-          worldUserLocationsById[user.id];
+      const recentLocationUsers =
+        (typeof locationName !== "string" || !isSingleObj) &&
+        Array.isArray(locationName)
+          ? locationName?.map((location) => {
+              const [, childLocation] = location.split("/");
+              const result = worldUsers.filter((user) => {
+                const userLocation: WithId<UserLocation> | undefined =
+                  worldUserLocationsById[user.id];
+
+                const userLastSeenIn = Object.keys(userLocation.lastSeenIn)[0];
+
+                return (
+                  userLastSeenIn &&
+                  (userLastSeenIn.includes(location) ||
+                    userLastSeenIn.includes(childLocation)) &&
+                  normalizeTimestampToMilliseconds(
+                    userLocation.lastSeenIn?.[location] ||
+                      userLocation.lastSeenIn?.[childLocation]
+                  ) > lastSeenThreshold
+                );
+              });
+
+              return result;
+            })
+          : worldUsers.filter((user) => {
+              const userLocation: WithId<UserLocation> | undefined =
+                worldUserLocationsById[user.id];
```

Comment:
looks complicated... can you explain the logic here?

Reply:
sure, previously `useRecentLocationUsers.ts` returned a single object that had `recentLocationUsers` for 1 exact location
now, since I need to fetch recent users for multiple locations ( events can be in different venues/rooms ) I need to fetch users per each location and return an array of users on each location
I thought of moving this logic to a separate file that would return this array, but decided to have a single hook in the end
tho still can move it out and separate logic

Reply:
As this hook previously worked with only one location and now works with multiple: can it be refactored in a way that the new implementation uses the old one?
I mean, for example for the hook `useRecentLocationUsers` to stay as it was (or with minor modifications) and to create a new one e.g. `useMultipleRecentLoationUsers` which somehow reuses logic from `useRecentLocationUsers`? I haven't thought of the exact way, it just sounds plausible to me.

Anyways, this code looks very complex, and other possible solutions: introduce more intermediate variables with descriptive names, reduce nesting, branching, extract logic to smaller functions, and so on.

Reply:
fixed in 36390a2
I've removed additional logic from this hook and created another one with array-like location name